src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java

```diff
@@ -31,24 +36,36 @@
   private final HttpServerHandler<ServerRequest, Response> handler;
   private final TraceContext.Extractor<ServerRequest> extractor;
   private final SpanNameProvider spanNameProvider;
+  private final RequestTagCustomizer requestTagCustomizer;
+  private final ResponseTagCustomizer responseTagCustomizer;
   private final Logger logger = LoggerFactory.getLogger(DefaultServerTracingHandler.class);
   @Inject
-  public DefaultServerTracingHandler(final HttpTracing httpTracing, final SpanNameProvider spanNameProvider) {
+  public DefaultServerTracingHandler(final HttpTracing httpTracing,
+                                     final SpanNameProvider spanNameProvider,
+                                     final RequestTagCustomizer requestTagCustomizer,
+                                     final ResponseTagCustomizer responseTagCustomizer) {
     this.tracing = httpTracing.tracing();
     this.handler = HttpServerHandler.<ServerRequest, Response>create(httpTracing, new ServerHttpAdapter());
     this.extractor = tracing.propagation().extractor((ServerRequest r, String name) -> r.getHeaders().get(name));
     this.spanNameProvider = spanNameProvider;
+    this.requestTagCustomizer = requestTagCustomizer;
+    this.responseTagCustomizer = responseTagCustomizer;
   }
 
   @Override
   public void handle(Context ctx) throws Exception {
     ServerRequest request = new ServerRequestImpl(ctx.getRequest());
-    final Span span = handler.handleReceive(extractor, request);
-    span.name(spanNameProvider.spanName(request, Optional.empty()));
+    final Span span = handler.handleReceive(extractor, request)
+                       .name(spanNameProvider.spanName(request, Optional.empty()));
+    for( Pair<String, String> kv: requestTagCustomizer.tags(request)) {
```

Comment:
handleReceive currently dodges parsing unless it can't. It would be neat to find a way to integrate what you have into the normal parser so that it can be controlled similarly. Otherwise you can consider guarding as parsing tags can sometimes add overhead you don't want in noop case.

ex. (if span.isNoop) // don't parse tags